test/controllers/submission/SubmissionControllerSpec.scala

```diff
@@ -18,189 +18,134 @@ package controllers.submission
+        val application = buildApplication()
+        running(application) {
+          val request = FakeRequest(POST, SubmissionController.submitDisclosure.url).withBody(Json.toJson(submissionDetails))
 
-      val argumentCaptor: ArgumentCaptor[NodeSeq]                      = ArgumentCaptor.forClass(classOf[NodeSeq])
-      val argumentCaptorSubmissionDetails: ArgumentCaptor[FileDetails] = ArgumentCaptor.forClass(classOf[FileDetails])
-      val argumentCaptorConversationId: ArgumentCaptor[ConversationId] = ArgumentCaptor.forClass(classOf[ConversationId])
+          val result = route(application, request).value
 
-      verify(mockFileDetailsRepository, times(1)).insert(argumentCaptorSubmissionDetails.capture())
-      verify(mockSubmissionConnector, times(1)).submitDisclosure(argumentCaptor.capture(), argumentCaptorConversationId.capture())(any[HeaderCarrier]())
+          status(result) mustBe OK
+          contentAsJson(result) mustBe Json.toJson(conversationId)
+        }
+      }
     }
 
-    "when a file is posted we transform it and trim and blank spaces on the node send it to the HOD expect return OK for Agent" in {
-      when(mockFileDetailsRepository.insert(any[FileDetails]()))
-        .thenReturn(Future.successful(true))
-      when(mockReadSubscriptionService.getContactInformation(any[String]())(any[HeaderCarrier](), any[ExecutionContext]()))
-        .thenReturn(Future.successful(Right(responseDetail)))
-      when(mockAgentReadSubscriptionService.getContactInformation(any[String]())(any[HeaderCarrier](), any[ExecutionContext]()))
-        .thenReturn(Future.successful(Right(agentResponseDetail)))
-      when(mockSubmissionConnector.submitDisclosure(any[NodeSeq](), any[ConversationId])(any[HeaderCarrier]()))
-        .thenReturn(Future.successful(HttpResponse(NO_CONTENT, "")))
-      when(mockXMLValidationService.validate(any[NodeSeq], any[String]))
-        .thenReturn(Right(basicXml))
-      val submission = basicXml
-
-      val request                = FakeRequest(POST, SubmissionController.submitDisclosure.url).withXmlBody(submission)
-      val result: Future[Result] = route(application, request).value
-
-      status(result) mustBe OK
-
-      val argumentCaptor: ArgumentCaptor[NodeSeq]                      = ArgumentCaptor.forClass(classOf[NodeSeq])
-      val argumentCaptorSubmissionDetails: ArgumentCaptor[FileDetails] = ArgumentCaptor.forClass(classOf[FileDetails])
-      val argumentCaptorConversationId: ArgumentCaptor[ConversationId] = ArgumentCaptor.forClass(classOf[ConversationId])
-
-      verify(mockFileDetailsRepository, times(1)).insert(argumentCaptorSubmissionDetails.capture())
-      verify(mockSubmissionConnector, times(1)).submitDisclosure(argumentCaptor.capture(), argumentCaptorConversationId.capture())(any[HeaderCarrier]())
-    }
+    "must treat file having size > maxNormalFileSizeBytes as large file" in {
+      forAll(arbitrary[SubmissionDetails]) { submissionDetails =>
```

Comment:
same as above - and also every where if we passing implicit arbitraries within `forAll`